stream.go

```diff
@@ -1465,6 +1480,10 @@ func (ss *serverStream) SetTrailer(md metadata.MD) {
 	if md.Len() == 0 {
 		return
 	}
+	if err := imetadata.Validate(md); err != nil {
+		logger.Errorf("stream: failed to set trailer, err: %v", err)
+		return
```

Comment:
Please delete the early `return` here and allow whatever bad behavior was happening before.